processing/src/main/java/org/apache/druid/query/UnnestDataSource.java

```diff
@@ -50,30 +48,26 @@ public class UnnestDataSource implements DataSource
   private final DataSource base;
   private final String column;
   private final String outputName;
-  private final LinkedHashSet<String> allowList;
 
   private UnnestDataSource(
       DataSource dataSource,
       String columnName,
-      String outputName,
-      LinkedHashSet<String> allowList
+      String outputName
   )
```

Comment:
I had expected you to change the allowList to a Filter but keep it on this object.  It looks like you are assuming that the filter passed into the call to make the cursor is good enough.  In the "best" case, you are planning things to attach the filter on the unnested thing to the other filters being included, in which case, as clint mentions, the entire filter is moving to a value matcher with absolutely nothing being pushed down.  In the worst case, some of the filters are being lost in planning and incorrect results can occur.  I haven't validated which one of these is likely occuring yet.

Given that Calcite very clearly puts the filter on the UNCOLLECT, you should be able to just attach a Filter object here instead of putting it on the query as a whole.  The logic inside of this should remain more or less the same as what it was before with the allow list, just, you are applying a value predicate to the dictionaries instead of building a bitset.

The filter used to make the cursor should continue to be passed down (though it should also have the filter that is attached here adjusted and pushed down as well).